homeassistant/components/generic/camera.py

```diff
@@ -166,4 +168,11 @@ def name(self):
 
     async def stream_source(self):
         """Return the source of the stream."""
-        return self._stream_source
+        if self._stream_source is None:
+            return None
+
+        try:
+            return self._stream_source.async_render()
+        except TemplateError as err:
+            _LOGGER.error("Error parsing template %s: %s", self._stream_source, err)
+            return self._stream_source
```

Comment:
Should we really return the non rendered template string on error? Maybe return `None` instead?

What is the expected outcome?

Reply:
You're right. Fixed.